phalcon/mailer.zep

```diff
@@ -0,0 +1,624 @@
+        }
+        else {
+            for key, value in parameters {
+                if property_exists(this,key) {
+                    this->setParam(key, value);
+                }
+            }
+        }
+    }
+
+    /**
+     * Set single parameter
+     *
+     * @param string  key
+     * @param type_key  value
+     * 
+     */
+    public function setParam(var key, var value) -> void {
+        let this->{key} = value;
+    }
+    
+
+    /**
+     * Connect to the server
+     * return status conection
+     * 
+     */
+    private function connect() -> boolean {
+        if strtolower(trim(this->secure)) == "ssl" {
+            let this->server = "ssl://" . this->server;
+        }
+        let this->conn = fsockopen(this->server, this->port, null, null, this->timeout);
+        if substr(this->getServerResponse(), 0, 3) != "220" { return false; }
+        return true;
+    }
+    
+    private function auth() -> boolean {
+        this->sendServer(this->conn, "HELO " . this->localhost . this->newline);
+        this->getServerResponse();
+        if strtolower(trim(this->secure)) == "tls" {
+            this->sendServer(this->conn, "STARTTLS" . this->newline);
+            if substr(this->getServerResponse(),0,3) != "220" { return false; }
+            stream_context_set_option(this->conn, "ssl", "verify_peer", false);
+            stream_context_set_option(this->conn, "ssl", "verify_peer_name", false);
+            stream_context_set_option(this->conn, "ssl", "allow_self_signed", true);
+            stream_socket_enable_crypto(this->conn, true, STREAM_CRYPTO_METHOD_TLS_CLIENT);
+            this->sendServer(this->conn, "HELO " . this->localhost . this->newline);
+            if substr(this->getServerResponse(), 0, 3) != "250" { return false; }
+        }
+        
+        if this->smtpAuth {
+            this->sendServer(this->conn, "AUTH LOGIN" . this->newline);
+            if (substr(this->getServerResponse(), 0, 3) != "334") { return false; }
+            this->sendServer(this->conn, base64_encode(this->username) . this->newline);
+            if (substr(this->getServerResponse(), 0, 3) != "334") { return false; }
+            this->sendServer(this->conn, base64_encode(this->password) . this->newline);
+            if (substr(this->getServerResponse(), 0, 3) != "235") { return false; }
+        }
+        return true;
+    }
+
+    /**
+     * Format date
+     * 
+     */
+    public function RFCDate() -> string {
+        var tz, tzs, result;
+        let tz = date('Z');
+        let tzs = (tz < 0) ? '-' : '+';
+        let tz = abs(tz);
+        let tz = (int)(tz/3600)*100 + (tz%3600)/60;
+        let result = sprintf("%s %s%04d", date("D, j M Y H:i:s"), tzs, tz);
+    
+        return result;
+    }
+
+    /**
+     * Set From Mail
+     *
+     * @param string  address
+     * @param string  name
+     * 
+     */
+    public function setFrom(address, name = "") -> void {
```

Comment:
public function setFrom(string! address, string! name = "") -> Mailer
{
    // ...
    return this;
}